APCAppCore/APCAppCore/UI/TabBarControllers/Profile/APCDownloadDataViewController.m

```diff
@@ -0,0 +1,187 @@
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+// OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+
+#import "APCDownloadDataViewController.h"
+#import "APCAppDelegate.h"
+#import "APCUser+Bridge.h"
+#import "NSDate+Helper.h"
+#import "APCAppCore.h"
+
+@interface APCDownloadDataViewController ()
+
+@property (nonatomic, strong) APCUser *user;
+@property (nonatomic, strong) UIToolbar *inputDoneBar;
+@property (nonatomic, strong) UIDatePicker *startDatePicker;
+@property (nonatomic, strong) UIDatePicker *endDatePicker;
+
+@end
+
+@implementation APCDownloadDataViewController
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    
+    [self setupTextFields];
+    [self updateDateTextFields];
+    [self setupNavAppearance];
+}
+
+- (void)didReceiveMemoryWarning {
+    [super didReceiveMemoryWarning];
+    // Dispose of any resources that can be recreated.
+}
+
+#pragma mark - Setup
+
+- (void)setupNavAppearance
+{
+    UIBarButtonItem  *backster = [APCCustomBackButton customBackBarButtonItemWithTarget:self action:@selector(back:) tintColor:[UIColor appPrimaryColor]];
+    [self.navigationItem setLeftBarButtonItem:backster];
+}
+
+- (void)setupTextFields {
+    [self.startTextField setFont:[UIFont appRegularFontWithSize:16.0f]];
+    [self.startTextField setTextColor:[UIColor appSecondaryColor1]];
+    [self.endTextField setFont:[UIFont appRegularFontWithSize:16.0f]];
+    [self.endTextField setTextColor:[UIColor appSecondaryColor1]];
+    
+    [self.startTextField setInputView:self.startDatePicker];
+    [self.endTextField setInputView:self.endDatePicker];
+    
+    if (_inputDoneBar == nil) {
+        _inputDoneBar = [[UIToolbar alloc] initWithFrame:CGRectMake(0, 0, self.view.bounds.size.width, 44)];
+        UIBarButtonItem *extraSpace = [[UIBarButtonItem alloc] initWithBarButtonSystemItem:UIBarButtonSystemItemFlexibleSpace target:nil action:nil];
+        UIBarButtonItem *done = [[UIBarButtonItem alloc] initWithTitle:@"Done" style:UIBarButtonItemStylePlain target:self action:@selector(updateDates:)];
+        [_inputDoneBar setItems:[[NSArray alloc] initWithObjects: extraSpace, done, nil]];
+    }
+    
+    self.startTextField.inputAccessoryView = _inputDoneBar;
+    self.endTextField.inputAccessoryView = _inputDoneBar;
+}
+
+#pragma mark - Getters
+
+- (UIDatePicker *) startDatePicker {
+    if (!_startDatePicker) {
+        _startDatePicker = [[UIDatePicker alloc] init];
+        NSDate *today = [NSDate date];
+        _startDatePicker.date = today.startOfYear;
```

Comment:
actually we don't want this to be the start of whatever the current year is--we want to default to giving them all their data, and 2015-01-01 is guaranteed to be before the start of any ResearchKit study (and should make it clear that this means they'd get all their data).